lib/msf/core/modules/external/templates/single_host_login_scanner.erb

```diff
@@ -24,7 +24,7 @@ class MetasploitModule < Msf::Auxiliary
   def run_host(ip)
     print_status("Running for #{ip}...")
     rhost = datastore.delete('RHOST')
-    datastore.remove_option('RHOST') if self.datastore.is_a?(Msf::DataStoreWithFallbacks)
```

Comment:
Are these necessary? Or would it be OK to remove these `remove_option` calls as the line below will overwrite potential RHOST values?

Reply:
Options are different to values; this was added for this bug fix where the `datastore` wasn't a `Msf::DataStoreWithFallbacks` instance: https://github.com/rapid7/metasploit-framework/pull/17792

Let me know if I've misunderstood though